test-cases/gutenberg/file.md

```diff
@@ -0,0 +1,219 @@
+
+# File Block - Test Cases
+##### TC001
+
+**Upload completes successfully from the device library**
+
+**Steps**
+- Tap on Choose a file on the Media Placeholder. 
+- Tap option "Choose from device" and choose a file. 
+
+**Expected Behavior**
+- Expect to see the upload indicator and the Download button should be dimmed.
+- Expect to not see a link or replace buttons in the toolbar.
+
+![Upload pending](../resources/local-file-upload-pending.png)
+- Expect to see file's name and an active Download button once the upload is complete.  
+- Expect the link and replace buttons to appear in the toolbar
+
+![Upload successful](../resources/local-file-upload-successful.png)
+
+--------------------------------------------------------------------------------
+
+##### TC002
+
+**Upload completes successfully from the WordPress Media library**
+
+**Steps**
+- Tap on Choose a file on the Media Placeholder. 
+- Tap option "WordPress Media Library" and choose a file. 
+
+**Expected Behavior**
+- Expect to see file's name  (without link)  and an active Download button.  
+
+![Upload successful](../resources/local-file-upload-successful.png)
+
+--------------------------------------------------------------------------------
+
+##### TC003
+
+**Upload fails from the device library**
+
+**Steps**
+-   Tap on Choose a file on the Media Placeholder or the edit button in the toolbar.
+-   Tap option ‘Choose from device’ and choose a file.
+-   Turn off the internet connection and wait for a moment.
+
+**Expected Behavior**
+-   You should see an error message and a dimmed Download button.
+
+![Upload failed](../resources/local-file-upload-failed.png)
+
+**Next Steps**
+
+-   Turn your internet connection back on.
+-   Tap on the File block itself.
+-   You should be asked to retry.
+
+**Expected Behavior**
+
+![Upload retry](../resources/local-file-upload-retry.png)
+
+**Next Step**
+-   Tap retry
+
+**Expected Behavior**
+-   The upload should continue.
+
+
+**Alternative Step**
+-   Tap Remove upload.
+
+**Expected Behavior**
+-   The placeholder should now be visible and the Download button and link should be cleared.
+--------------------------------------------------------------------------------
+##### TC004
+
+**Stop the file upload before it completes**
+
+**Steps**
+
+-   Tap on Choose a file on the Media Placeholder or edit button in the toolbar.
+-   Tap option ‘Choose from device’ and choose a file.
+-   While the upload is in progress tap on the File Block.
+
+**Expected Behavior**
+
+-   You should see the below message:
+![Upload stopped](../resources/local-file-upload-stop.png)
+
+**Next Step**
+
+-   Tap Yes
+
+**Expected Behavior**
+
+-   You should see that File block is cleared and the placeholder component is now visible.
+![Upload placeholder](../resources/local-file-upload-placeholder.png)
+
+--------------------------------------------------------------------------------
+##### TC005
+
+**Close the post with an ongoing file upload**
+
```

Comment:
I would also add the test case of closing the editor and opening it again while a file is being updated.
Expected:
- The block should detect the ongoing upload and show the progress bar.

Reply:
Done 30d7541